scripts/extensions/sams/src/components/sets/setListItem.tsx

```diff
@@ -0,0 +1,88 @@
+// External Modules
+import * as React from 'react';
+
+// Types
+import {ISetItem, IStorageDestinationItem, SET_STATE} from '../../interfaces';
+
+// UI
+import {Badge, IconButton} from 'superdesk-ui-framework/react';
+import {ListItem, ListItemActionMenu, ListItemBorder, ListItemColumn, ListItemRow} from '../../ui/list';
+
+interface IProps {
+    set: ISetItem;
+    storageDestination?: IStorageDestinationItem;
+    selected?: boolean;
+    onClick(set: ISetItem): void;
+    deleteSet?(set: ISetItem): void;
+    editSet?(set: ISetItem): void;
+}
+
+export class SetListItem extends React.PureComponent<IProps> {
+    constructor(props: IProps) {
+        super(props);
+
+        this.onItemClick = this.onItemClick.bind(this);
+        this.onEdit = this.onEdit.bind(this);
+        this.onDelete = this.onDelete.bind(this);
+    }
+
+    onItemClick(event: React.MouseEvent<HTMLDivElement>) {
+        if (this.props.onClick) {
+            event.stopPropagation();
+            this.props.onClick(this.props.set);
+        }
+    }
+
+    onEdit(event: any) {
+        if (this.props.editSet) {
+            event.stopPropagation();
+            this.props.editSet(this.props.set);
+        }
+    }
+
+    onDelete(event: any) {
+        if (this.props.deleteSet) {
+            event.stopPropagation();
+            this.props.deleteSet(this.props.set);
+        }
+    }
+
+    render() {
+        const {set, storageDestination} = this.props;
+        const storageDestinationText = storageDestination == null ?
+            '' :
+            `${storageDestination._id} / ${storageDestination.provider}`;
```

Comment:
I wonder if we will need translations here in the future. For example I can see now `Default` as the id and that's probably something we want to translate.

Reply:
This is actually defined in the `settings.py` from the SAMS API, so this will be instance specific.